src/cipher/mod.rs

```diff
@@ -0,0 +1,359 @@
+use dusk_plonk::jubjub::AffinePoint;
+use dusk_plonk::prelude::*;
+use hades252::{ScalarStrategy, Strategy, WIDTH};
+
+use std::io;
+
+/// Maximum number of scalars allowed per message
+pub const MESSAGE_CAPACITY: usize = 2;
+
+/// Number of scalars used in a cipher
+pub const CIPHER_SIZE: usize = MESSAGE_CAPACITY + 1;
+
+/// Bytes consumed on serialization of the poseidon cipher
+pub const ENCRYPTED_DATA_SIZE: usize = CIPHER_SIZE * 32;
+
+/// Plonk gadget for Poseidon encryption
+pub mod gadget;
```

Comment:
I'd probably rename the mod. `Gadget` is too generic imo. and we have more than one on this repo, so it should be clear which one it is.

Reply:
Renamed to `zk`